Software/VelvetApp/Core/Src/rfid.c

```diff
@@ -0,0 +1,184 @@
+		for(ucJ = 0; ucJ < 8; ucJ++){
+			if(uiCrcValue & 0x0001) uiCrcValue = (uiCrcValue >> 1) ^ POLYNOMIAL;		   	
+			else uiCrcValue = (uiCrcValue >> 1);
+		}
+ 	}
+	return uiCrcValue;
+}
+
+void setCRC16 (uint8_t* buf){
+	uint16_t crc = uiCrc16Calc(buf, buf[0]-1);
+	buf[buf[0]-1] = crc&0xff;
+	buf[buf[0]] = (crc&0xff00)>>8;
+}
+
+void getReaderInfo (uint8_t* buf){
+	buf[0] = 4;
+	buf[1] = 0xff;
+	buf[2] = 0x21;
+	setCRC16(buf);	
+}
+
+void writeCMDToBuf(uint8_t* buf, uint8_t cmdNum){
+	switch(cmdNum){
+		case getReaderInfoCMD:
+			buf[0] = 4;
+			buf[1] = 0xff;
+			buf[2] = cmdNum;			
+		break;
+		case inventoryG2CMD:
+			buf[0] = 13;
+			buf[1] = deviceAddress;
+			buf[2] = cmdNum;
+			buf[3] = 4;//QValue
+			buf[4] = 0;//Session S0
+			buf[5] = 1;//MaskMem
+			buf[6] = 0;//MaskAdr
+			buf[7] = 0;//MaskAdr
+			buf[8] = 0;//MaskLen
+			buf[9] = 0;//MaskData
+			buf[10] = 0;//AdrTID
+			buf[11] = 0;//LenTID
+		break;
+	}
+	setCRC16(buf);
+	currentCmdRFID = buf[2];	
+}
+
+void sendCmd(uint8_t* buf, uint8_t cmdNum){
+	writeCMDToBuf(buf, cmdNum);
+	HAL_UART_Transmit_IT(&huart6,buf,buf[0]+1);	
+}
+
+void rfidInit(void){
+	HAL_GPIO_WritePin(RFID_EN_GPIO_Port, RFID_EN_Pin, GPIO_PIN_SET);
+	osDelay(2000);
+	sendCmd(outputBuffer, getReaderInfoCMD);
+	//writeCMDToBuf(outputBuffer, getReaderInfoCMD);
+	//->getReaderInfo(outputBuffer);
+	//HAL_UART_Transmit_IT(&huart6,outputBuffer,outputBuffer[0]+1);
+	//receiveStart = 1;
+}
+
+int8_t checkBufCRC(uint8_t* buf){
+	uint16_t crcReceive;
+	
+	uint16_t crcCalc = uiCrc16Calc(buf, buf[0]-1);	
+	crcReceive = (buf[buf[0]]<<8)|buf[buf[0]-1];
+	if(crcCalc == crcReceive) return RFID_OK;
+	return RFID_ERROR;
+}
+
+uint8_t isTagsEqual(uint8_t* newTag, uint8_t* currentTag){
+	uint8_t i=0;
+	
+	for(i=0;i<6;i++){
+		if(newTag[i]!=currentTag[i]) return 0;
+	}
+	return 1;
+}
+
+void copyTag(uint8_t* srcTag, uint8_t* destTag){
+	uint8_t i=0;
+	
+	for(i=0;i<6;i++) destTag[i]=srcTag[i];
+}
+void readEPCData(uint8_t* buf){
+	uint8_t i=0, j, tagsNum = buf[5], tempBuf[7];	
+	
+	while(tagsNum>0){
+		for(j=0;j<7;j++) tempBuf[j] = buf[10+j];
+		for(j=0;j<6;j++) tagsBuffer[i][j] = ((tempBuf[j]&0x0f)<<4)|((tempBuf[j+1]&0xf0)>>4);		
+		tagsNum--;
+		i++;
+	}
+	if(buf[5]>0){		
+		if(isTagsEqual(&tagsBuffer[i-1][0], currentTag) == 0){
+			//if(adcConversionInProcess==0){
+				copyTag(&tagsBuffer[i-1][0], currentTag);
+				//osSemaphoreRelease(readWeightSemHandle);
+			//}
+		}			
+	}
+}
+
+void readRfidResponse(uint8_t* buf){
+	if(checkBufCRC(buf) == RFID_OK) {
+		if(currentCmdRFID == buf[2]){
+			switch(buf[2]){
+				case getReaderInfoCMD:
+					deviceAddress = buf[1];
+					rfidInitState = RFID_OK;
+				break;
+				case inventoryG2CMD:
+					if((buf[3] == 1)&&(buf[5] > 0)) readEPCData(buf);
+				break;
+			}
+		}
+	}	
+}
+
+void rfidReInit(void){
+	HAL_GPIO_WritePin(RFID_EN_GPIO_Port, RFID_EN_Pin, GPIO_PIN_RESET);
+	osDelay(1000);
+	receiveStage = 0;
+	rfidInit();
+}
+
+void readRfid(void *argument)
```

Comment:
Предлагаю переименовать в 
```suggestion
void rfid_worker(void *argument)
```

Reply:
+